### PR TITLE
Add remote zip support to mc ls/cp/cat

### PR DESCRIPTION
**Pull request by klauspost**
All remote zip support for mc operations.

```
λ mc ls --zip play/test/repro-pack.zip/data/mindev/data2/xl1/src3/src2/bytes/bytes.go/
[2022-02-09 16:38:59 CET]  14KiB STANDARD part.1
[2022-02-09 16:38:59 CET] 1.1KiB STANDARD xl.meta

λ mc cat --zip play/test/repro-pack.zip/data/mindev/data2/xl1/src3/src2/bytes/bytes.go/xl.meta
{"version":"1.0.1 ...

λ mc cp --zip play/test/repro-pack.zip/data/mindev/data2/xl1/src3/src2/bytes/bytes.go/xl.meta .
.../xl1/src3/src2/bytes/bytes.go/xl.meta:  1.06 KiB / 1.06 KiB [========================================] 1.21 KiB/s 0s

λ mc cp --recursive --json --zip play/test/repro-pack.zip/data/mindev/data2/xl1/src3/src2/bytes/bytes.go/ .
{
 "status": "success",
 "source": "play/test/repro-pack.zip/data/mindev/data2/xl1/src3/src2/bytes/bytes.go/part.1",
 "target": "part.1",
 "size": 14625,
 "totalCount": 2,
 "totalSize": 0
}
{
 "status": "success",
 "source": "play/test/repro-pack.zip/data/mindev/data2/xl1/src3/src2/bytes/bytes.go/xl.meta",
 "target": "xl.meta",
 "size": 1084,
 "totalCount": 2,
 "totalSize": 0
}

λ mc ls --recursive --zip play/test/repro-pack.zip/data/mindev/data2/xl1/src3/src2/bytes/
[2022-02-09 16:38:59 CET] 1.1KiB STANDARD boundary_test.go/part.1
[2022-02-09 16:38:59 CET] 1.1KiB STANDARD boundary_test.go/xl.meta
[2022-02-09 16:38:59 CET] 7.2KiB STANDARD buffer.go/part.1
[2022-02-09 16:38:59 CET] 1.1KiB STANDARD buffer.go/xl.meta
...
```

### CODE REVIEW

**Comment by klauspost** [comment]
On purpose not adding examples to keep it a bit low key for now. Doing a blog post where I will cover it.

**Review by harshavardhana — APPROVED** [review]

File: cmd/ls-main.go [hunk 1]
```diff
@@ -169,15 +173,27 @@ func checkListSyntax(ctx context.Context, cliCtx *cli.Context) ([]string, bool,
...
 	withOlderVersions := cliCtx.Bool("versions")
 	isSummary := cliCtx.Bool("summarize")
+	listZip := cliCtx.Bool("zip")
 
 	timeRef := parseRewindFlag(cliCtx.String("rewind"))
 	if timeRef.IsZero() && withOlderVersions {
 		timeRef = time.Now().UTC()
 	}
 
+	if listZip && withOlderVersions {
```
Comment (vadmeste) [inline cmd/ls-main.go:183]: listZip && (withOlderVersions || !timeRef.IsZero())

File: cmd/cp-main.go [hunk 2]
```diff
@@ -641,7 +667,6 @@ func mainCopy(cliCtx *cli.Context) error {
 
 	// check 'copy' cli arguments.
 	checkCopySyntax(ctx, cliCtx, encKeyDB, false)
```
Comment (vadmeste) [inline cmd/cp-main.go:669]: Can you error out if --rewind and --zip are passed ?

**Review by vadmeste — COMMENTED** [review]
Two comments

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**